### PR TITLE
Added resource routes

### PR DESCRIPTION
**Pull request by umbreak**
Related to https://github.com/BlueBrain/nexus/issues/1771

### CODE REVIEW

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                }
+              }
+            )
+          }
+        )
+      }
+    )
+
+  private def fetchProject(orgUuid: UUID, projectUuid: UUID): Directive1[ProjectResource] =
+    onSuccess(projects.fetch(projectUuid).runToFuture).flatMap {
```
Comment (imsdu) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala:167]: Only the ref is interesting here, we should create a method findRefByUuid that return the ref from the uuids.
This way, we only aim for the cache once.
It is something we will need for other resources so we could put it in an object

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+      },
+      // Create a resource without schema nor id segment
+      (post & noParameter("rev") & pathEndOrSingleSlash & entity(as[Json])) { source =>
+        operationName(s"$prefixSegment/resources/{org}/{project}") {
+          authorizeFor(AclAddress.Project(ref), resourcePermissions.write).apply {
+            emit(StatusCodes.Created, resources.create(ref, resourceSchema, source).map(_.void))
+          }
+        }
+      },
+      idSegment { schemaSegment =>
```
Comment (imsdu) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala:102]: If we could parse to something like a NoSegment, couldn't we parse the two segments at the same time and make the routes a little simpler ?

**Review by imsdu — CHANGES_REQUESTED** [review]

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+                      emit(resources.events(offset))
+                    }
+                  }
+                }
+              }
+            },
+            projectRef { ref =>
+              routesForProject(ref)
+            },
+            (uuid & uuid) { (orgUuid, projectUuid) =>
```
Comment (imsdu) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala:71]: We could have a directive projectRefFromUuid

**Review by imsdu — CHANGES_REQUESTED** [review]

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+            authorizeFor(AclAddress.Project(ref), events.read).apply {
+              lastEventId { offset =>
+                emit(resources.events(ref, offset).leftWiden[ResourceRejection])
+              }
+            }
+          }
+        }
+      },
+      // Create a resource without schema nor id segment
+      (post & noParameter("rev") & pathEndOrSingleSlash & entity(as[Json])) { source =>
```
Comment (bogdanromanx) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala:95]: I think currently we also allow `POST /resources/{org}/{proj}/_`

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+  private val resourceSchema = IriSegment(schemas.resources)
+
+  def routes: Route =
+    baseUriPrefix(baseUri.prefix) {
+      extractCaller { implicit caller =>
+        pathPrefix("resources") {
+          concat(
+            // SSE resources
+            (pathPrefix("events") & pathEndOrSingleSlash) {
```
Comment (bogdanromanx) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala:57]: we seem to be missing `GET /resources/{org}/events`

File: delta/app/src/test/resources/organizations/org-resource.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "@context": [
-    "https://bluebrain.github.io/nexus/contexts/resource.json",
-    "https://bluebrain.github.io/nexus/contexts/organizations.json"
+    "https://bluebrain.github.io/nexus/contexts/organizations.json",
+    "https://bluebrain.github.io/nexus/contexts/resource.json"
```
Comment (bogdanromanx) [inline delta/app/src/test/resources/organizations/org-resource.json:4]: we should standardize on plural

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala (hunk 2 above)
Comment (umbreak) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala:102]: I think it is perfectly fine as is. More abstraction is just going to make it more complicated. Besides now it is in order:

```scala
idSegment { schemaSegment =>
   // something to do after only the schema segment has been consumed
   idSegment { idSegment =>
      // something to do after both schema and id segment have been consumed
   }
}
```

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala (hunk 3 above)
Comment (umbreak) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala:71]: Yes, but would it make things clearer? I find this line already very explicit :)

**Review by bogdanromanx — CHANGES_REQUESTED** [review]
Looks good, some small things...

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala (hunk 2 above)
Comment (imsdu) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala:102]: I just find the route a little complicated to understand and am trying to find a way to make it a little simpler. But yes, my solution may not be a good one

File: delta/app/src/test/resources/organizations/org-resource.json (hunk 6 above)
Comment (umbreak) [reply delta/app/src/test/resources/organizations/org-resource.json:4]: Notice that resource is just the nexus metadata.  It is not a domain resource as organizations, projects, schemas,  etc. So maybe we can call it differently. Ideas?

Comment (bogdanromanx) [reply delta/app/src/test/resources/organizations/org-resource.json:4]: hmmm, good point; maybe renaming it to `metadata.json`?

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala (hunk 4 above)
Comment (umbreak) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala:95]: Yes, this is already supported: https://github.com/BlueBrain/nexus/blob/74b8a7ec73bd1866e8c65b6b0a8cbe0529b6b3a6/delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala#L106

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala [hunk 7]
```diff
@@ -65,109 +62,115 @@ final class ResourcesRoutes(
...
-          (post & noParameter("rev") & pathEndOrSingleSlash) {
-            operationName(s"$prefixSegment/resources/{org}/{project}/{schema}") {
-              authorizeFor(AclAddress.Project(ref), resourcePermissions.write).apply {
-                entity(as[Json]) { source =>
-                  emit(StatusCodes.Created, resources.create(ref, schemaSegment, source).map(_.void))
+            // SSE resources for all events belonging to an organization
+            ((label | orgLabelFromUuidLookup) & pathPrefix("events") & pathEndOrSingleSlash) { org =>
+              get {
+                operationName(s"$prefixSegment/resources/{org}/events") {
+                  authorizeFor(AclAddress.Root, events.read).apply {
```
Comment (bogdanromanx) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala:69]: I think the scope of the permission should be `AclAddress.Organization(org)`

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

**Review by bogdanromanx — APPROVED** [review]

**Review by imsdu — APPROVED** [review]

**Pull request merged**